### PR TITLE
Remove differences of the code between old AnimationTree and AnimationMixer

### LINKED ISSUE #85640: AnimationNodeStateMachinePlayback.travel() gets stuck for nested state machines
### Godot version

4.2

### System information

Linux

### Issue description

A nested state machine isn't traveling to its animation in `_ready()` in 4.2 as it was in 4.1.3.

In the ready function, we make the `root` state machine travel to the state of the `Stages` nested state machine. Then we immediately call the `Stages` nested state machine to travel to `Stage2`. This works fine in Godot v4.1.3, but not in Godot v4.2, where it seems a delay is first needed for the second travel animation to work.

image
image

### Steps to reproduce

Run the MRP below in 4.1.3 and 4.2.

In 4.1.3, the animation tree will reach `Stage2` and show Finished on the screen.

In 4.2, the animation tree will only reach `Stage1` and show Stuck on the screen.

### Minimal reproduction project

[MRP-AnimationNodeStateMachinePlayback.zip](https://github.com/godotengine/godot/files/13533132/MRP-AnimationNodeStateMachinePlayback.zip)

### PR DESCRIPTION
**Pull request by TokageItLab**
- Fixes https://github.com/godotengine/godot/issues/85640

A few lines were overlooked during those porting, sorry. I think the problem only occurred in complex use cases.

It would be helpful if someone could test on other complex trees besides the above issues.

### CODE REVIEW

**Review by fire — APPROVED** [review]
p_test_only does seem like dead code. The changes here suggest that it is always false. Only did a github review and not via a comprehensive test suit.

Edited:

It's a bit tricky to understand though. Do you have a test case where it fails? One of the attached issues mentioned that this fixes things by not creating empty nodes.

**Comment by TokageItLab** [comment]
`test_only` was currently dead code, but in 4.1 it works to process the AnimationNode virtually for estimating remaining time. However, `test_only` uses only complex nested/root StateMachines, it is difficult to cover test cases.

File: scene/animation/animation_tree.cpp [hunk 1]
```diff
@@ -565,12 +565,12 @@ bool AnimationTree::_blend_pre_process(double p_delta, int p_track_count, const
...
-			root_animation_node->_pre_process(&process_state, pi);
+			root_animation_node->_pre_process(&process_state, pi, false);
 			started = false;
-		} else {
-			pi.time = p_delta;
-			root_animation_node->_pre_process(&process_state, pi);
 		}
+		pi.seeked = false;
+		pi.time = p_delta;
+		root_animation_node->_pre_process(&process_state, pi, false);
```
[the remark this answers is not in the document]
Comment (TokageItLab) [reply scene/animation/animation_tree.cpp:573]: Before 4.1, it handled both the current position immediately after playback and the delta afterwards, so the purpose is to revert the behavior for now. If there is a problem, it should already be a problem with AnimationTree in 4.1, so this probably won't be a problem.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by YuriSizov** [comment]
Given that https://github.com/godotengine/godot/pull/86644 addresses my concern from one of the changes here, I think they should be cherry-picked together. So setting this aside for now since https://github.com/godotengine/godot/pull/86644 is not merged yet.

**Pull request merged**